packages/py-moose-lib/moose_lib/dmv2/olap_dictionary.py

```
@@ -0,0 +1,592 @@
+            raise ValueError("At least one key argument is required")
+        expected = len(self.config.primary_key)
+        if len(keys) != expected:
+            raise ValueError(
+                f"Expected {expected} key argument(s) to match primary_key "
+                f"{self.config.primary_key}, got {len(keys)}"
+            )
+        if len(keys) > 1:
+            return f"tuple({', '.join(str(k) for k in keys)})"
+        return str(keys[0])
+
+    def get(self, attr: str, *keys) -> str:
```

Ruff (0.15.12)

[warning] Missing type annotation for `*keys`

(ANN002)